folktables/utils/download_utils.py

```diff
@@ -0,0 +1,135 @@
+import concurrent.futures
+import os
+import pathlib
+
+import requests
+
+from folktables import exceptions
+
+
+def download_file(url, download_path):
+    """Makes a GET request to the specified URL and saves the
+    contents of the response to the specified path.
+
+    Parameters
+    ----------
+    url : str
+        URL from where the data will be downloaded.
+    download_path : pathlib.Path
+        Path to where the downloaded content will be stored.
+
+    Returns
+    -------
+    download_path : str
+        Path to where the content is stored.
+
+    Raises
+    ------
+    exceptions.FileDownloadError
+        This exception is raised if we get an error from our HTTP request
+        (i.e., the status code we get from the response is not 200).
+    """
+    response = requests.get(url)
+
+    if response.status_code != 200:
```

Comment:
You could also handle this via one line: https://requests.readthedocs.io/en/latest/api/#requests.Response.raise_for_status

Reply:
I can certainly use `raise_for_status()`. Now the question is: **should we handle catch the `HTTPError` it raises when the status code is not within the 200 range, or should we just let this exception bubble up to the user?**

If we handle the exception to re-raise one of our exceptions (i.e., `FileDownloadError`) the code will look like this:
```py
    try:
        response.raise_for_status()
    except requests.exceptions.HTTPError:
        raise exceptions.FileDownloadError(
            f'Failed to download the data from: {url}\n'
            f'The HTTP request returned a {response.status_code} status code.'
        )
```

Reply:
Good question about exception handling, I'm also not 100% sure about this. I tend to the "let things bubble up" solution here for the following reason: the download can fail due to a variety of causes (DNS resolution error, timeout, bad response code, etc.). Each of these may result in a different exception. At the user level, it's probably helpful to know exactly why the download failed, so we shouldn't hide this information. If we catch these exceptions and then wrap them in our own exceptions, that's adding a lot of complexity that we could avoid by just letting the error bubble up.

An intermediate solution would be to use the "raise ... from" syntax to forward the requests exception as an inner exception, e.g., see:

https://stackoverflow.com/questions/24752395/python-raise-from-usage

But if we're really just bubbling up the exception and don't do any additional error handling here, probably the simple thing (i.e., nothing except `raise_for_status()`) may be the right thing to do?

Reply:
Ok perfect. Then I'll change the code to let the exception just bubble up.